framework/src/main/java/org/tron/core/zen/ZksnarkInitService.java

```diff
@@ -0,0 +1,56 @@
+package org.tron.core.zen;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.tron.common.zksnark.JLibrustzcash;
+import org.tron.common.zksnark.LibrustzcashParam;
+import org.tron.core.exception.ZksnarkException;
+
+@Slf4j(topic = "API")
+public class ZksnarkInitService {
+
+  static {
+    librustzcashInitZksnarkParams();
```

Comment:
When is this method called? The original logic was to call it when the HTTP service is enabled. What would happen if this method is called when the HTTP service is not enabled?

Reply:
For shielded

Reply:
Prepare to use `@DependsOn("fullNodeHttpApiService") `to keep the original logic

Reply:
@Federico2014 PTAL.